libraries/AP_Param/AP_Param.cpp

```diff
@@ -1160,12 +1160,21 @@ void AP_Param::save_sync(bool force_save)
 */
 void AP_Param::save(bool force_save)
 {
-    struct param_save p;
+    struct param_save p, p2;
     p.param = this;
     p.force_save = force_save;
+    if (save_queue.peek(p2) &&
```

Comment:
I think it's fine, but it's worth noting that this depends upon grabbing a mutex from the save thread here to do this check. I think it will be fine, but I haven't fully followed the lock on the scheduler semaphore.

Reply:
yes, I think it is ok, as we are grabbing/releasing in the right order